### PR TITLE
Uploading a sample

### PR DESCRIPTION
**Pull request by normesta**
## Purpose
Sample contributed by the CSS team. It's been super well tested. I'm a writer on the Azure Storage team and intend to source this code in an article.
* ...

## Does this introduce a breaking change?

```
[ ] Yes
[X ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Log into Azure by using the "Connect-AzAccount". Add the name of a storage account and a resource group to the script and then run the script at a Windows PowerShell window.

### CODE REVIEW

**Comment by normesta** [comment]
@mikefrobbins are you the appropriate person to review this sample / merge? I don't seem to have merge power in this repo.

Thx!

Norm E.

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# note: this retrieves all of the blobs in each container in one command.
+#       Run the Connect-AzAccount cmdlet to connect to Azure.
+#       Requests that are sent as part of this tool will incur transactional costs.
+#
+
+$containerstats = @()
+
+# Provide the name of your storage account and resource group
+$storage_account_name = "<name-of-your-storage-account>"
+$resource_group = "<name-of-your-resource-group"
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:13]:
```suggestion
<#
.SYNOPSIS
Calculates the total size of blobs in all containers in a specified Azure storage account.

.DESCRIPTION
Before running this script, ensure you have:
- A storage account created
- At least one container in the storage account
- Uploaded some blobs into the container

.EXAMPLE
.\Get-AzureStorageAccountBlobSize.ps1 -StorageAccountName mystorageaccount -ResourceGroupName myResourceGroup

.NOTES
This script incurs transactional costs for Azure requests.
#>

[CmdletBinding()]
param (
    [ValidateNotNullOrEmpty()]
    [string]$ResourceGroupName = '<name-of-your-resource-group>',
    
    [ValidateNotNullOrEmpty()]
    [string]$StorageAccountName = '<name-of-your-storage-account>'
)

$containerstats = @()
```

Use comment-based help and parameterize the script. Without `[CmdletBinding()]`, the `Write-Verbose` statements will never be displayed due to the script not having a **Verbose** parameter.

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        Name = $container 
+        TotalBlobCount = $total_blob_count 
+        TotalBlobUsageinGB = $total_usage/1GB
+        SoftDeletedBlobCount = $soft_delete_count
+        SoftDeletedBlobUsageinGB = $soft_delete_usage/1GB
+		SnapshotCount = $snapshot_count
+		SnapshotUsageinGB = $snapshot_usage/1GB
+		VersionCount = $version_count
+		VersionUsageinGB = $version_usage/1GB
+      }
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:94]:
```suggestion
      $containerstats += [PSCustomObject] @{ 
        Name = $container 
        TotalBlobCount = $total_blob_count 
        TotalBlobUsageinGB = $total_usage/1GB
        SoftDeletedBlobCount = $soft_delete_count
        SoftDeletedBlobUsageinGB = $soft_delete_usage/1GB
	SnapshotCount = $snapshot_count
	SnapshotUsageinGB = $snapshot_usage/1GB
	VersionCount = $version_count
	VersionUsageinGB = $version_usage/1GB
      }
```
Format the code with the same alignment.

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  -Name $storage_account_name
+$Ctx = $storageAccount.Context
+
+$container_continuation_token = $null
+do {
+  $containers = Get-AzStorageContainer -Context $Ctx -MaxCount 5000 -ContinuationToken $container_continuation_token    
+  $container_continuation_token = $null;
+
+  if ($containers -ne $null)
+  {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:27]:
```suggestion
  if ($containers -ne $null) {
```
Use OTBS (One true brace style) so your `if` statements are formatted the same as the code for the `do` loops.

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    $container_continuation_token = $containers[$containers.Count - 1].ContinuationToken
+    
+    for ([int] $c = 0; $c -lt $containers.Count; $c++)
+    {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:31]:
```suggestion
    for ([int] $c = 0; $c -lt $containers.Count; $c++) {
```

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+             $snapshot_count = 0 
+             $snapshot_usage = 0
+      $blob_continuation_token = $null
+      
+      do {
+        $blobs = Get-AzStorageBlob -Context $Ctx -IncludeDeleted -IncludeVersion -Container $container -ConcurrentTaskCount 100 -MaxCount 5000 -ContinuationToken $blob_continuation_token
+        $blob_continuation_token = $null;
+        
+        if ($blobs -ne $null)
+        {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:49]:
```suggestion
        if ($blobs -ne $null) {
```

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+          $blob_continuation_token = $blobs[$blobs.Count - 1].ContinuationToken
+          
+          for ([int] $b = 0; $b -lt $blobs.Count; $b++)
+          {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:53]:
```suggestion
          for ([int] $b = 0; $b -lt $blobs.Count; $b++) {
```

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            $total_blob_count++
+            $total_usage += $blobs[$b].Length
+            
+            if ($blobs[$b].IsDeleted)
+            {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:58]:
```suggestion
            if ($blobs[$b].IsDeleted) {
```

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+              $soft_delete_count++
+              $soft_delete_usage += $blobs[$b].Length
+            }
+            
+            if ($blobs[$b].SnapshotTime -ne $null)
+            {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:64]:
```suggestion
            if ($blobs[$b].SnapshotTime -ne $null) {
```

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+              $snapshot_count++
+              $snapshot_usage+= $blobs[$b].Length
+            }
+            
+            if ($blobs[$b].VersionId -ne $null)
+            {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:70]:
```suggestion
            if ($blobs[$b].VersionId -ne $null) {
```

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+              $version_count++
+              $version_usage += $blobs[$b].Length
+            }
+          }
+          
+          If ($blob_continuation_token -ne $null)
+          {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:77]:
```suggestion
          if ($blob_continuation_token -ne $null) {
```
Use consistent casing.

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+		SnapshotCount = $snapshot_count
+		SnapshotUsageinGB = $snapshot_usage/1GB
+		VersionCount = $version_count
+		VersionUsageinGB = $version_usage/1GB
+      }
+    }
+  }
+  
+  If ($container_continuation_token -ne $null)
+  {
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:99]:
```suggestion
  if ($container_continuation_token -ne $null) {
```

File: storage/calculate-container-size/calculate-container-sizes-in-account.ps1 [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+      Write-Verbose "Processing container : $container"
+      $total_usage = 0
+      $total_blob_count = 0
+      $soft_delete_usage = 0
+      $soft_delete_count = 0
+             $version_usage = 0
+             $version_count = 
+             $snapshot_count = 0 
+             $snapshot_usage = 0
+      $blob_continuation_token = $null
```
Comment (mikefrobbins) [inline storage/calculate-container-size/calculate-container-sizes-in-account.ps1:42]:
```suggestion
      $soft_delete_count = 0
      $version_usage = 0
      $version_count = 
      $snapshot_count = 0 
      $snapshot_usage = 0
      $blob_continuation_token = $null
```
Format the code with proper alignment.

**Review by mikefrobbins — CHANGES_REQUESTED** [review]
No logic changes since it's well tested, but several best practice, formatting, and usability changes requested.

**Comment by normesta** [comment]
Thank you @mikefrobbins! I committed your suggestions and tested the updates and all works great.

**Review by mikefrobbins — APPROVED** [review]
LGTM

**Pull request merged**